source/delimiter-case.d.ts

```diff
@@ -87,13 +54,9 @@ const rawCliOptions: OddlyCasedProperties<SomeOptions> = {
 
 @category Change case
 @category Template literal
-*/
-export type DelimiterCase<Value, Delimiter extends string> = string extends Value ? Value : Value extends string
-	? StringArrayToDelimiterCase<
-	SplitIncludingDelimiters<Value, WordSeparators | UpperCaseCharacters>,
-	true,
-	WordSeparators,
-	UpperCaseCharacters,
-	Delimiter
-	>
+ */
+export type DelimiterCase<Value, Delimiter extends string, Options extends SplitWordsOptions = {splitOnNumber: true}> = Value extends string
```

Comment:
Here we
1. `'hereWeGo'` -> SplitWords -> `['here', 'We', 'Go']`
2. `['here', 'We', 'Go']` -> DelimiterCaseFromArray ->  `'#here#We#Go'`
3. `'#here#We#Go'` -> RemoveFirstLetter ->  `'here#We#Go'`
4. `'here#We#Go'` -> Lowercase -> `'here#we#go'`